### PR TITLE
[Add]:Client updation. deletion, new token generation.

### PR DESCRIPTION
**Pull request by nishihere19**

### CODE REVIEW

File: src/controllers/clientController.ts [hunk 1]
```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
...
+            'You do not have the required permissions to delete the client.'
+        });
+
+    //delete client
+    const result = await prisma.client.delete({
+      where: {
+        clientId: clientId
+      }
+    });
+    if (!result) return res.status(400).json({ message: 'Client not found' });
```
Comment (siva2204) [inline src/controllers/clientController.ts:168]: client can't be null here, no need of this check ig

File: src/controllers/clientController.ts [hunk 2]
```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
...
+
+//to delete a client
+export const deleteClient = async (
+  req: Request,
+  res: Response
+): Promise<unknown> => {
+  try {
+    const { clientId } = req.body;
+    if (!clientId)
+      return res.status(422).json({ message: 'Client Id is required.' });
```
Comment (siva2204) [inline src/controllers/clientController.ts:144]: 400 is best?

File: src/controllers/clientController.ts [hunk 3]
```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
...
+  try {
+    const { clientId, name, homePageUrl, callbackUrl, description } = req.body;
+    if (!clientId)
+      return res.status(422).json({ message: 'Client Id is required.' });
+    const user: any = req.user;
+    // fetch client data
+    const client = await prisma.client.findUnique({
+      where: {
+        clientId: clientId
+      }
```
Comment (siva2204) [inline src/controllers/clientController.ts:189]: select{userid:true}
better to make middleware to check if the client exists no?

File: src/controllers/clientController.ts [hunk 4]
```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
...
+        clientId: clientId
+      },
+      data: {
+        name: name,
+        homePageUrl: homePageUrl,
+        redirectUri: callbackUrl,
+        description: description
+      }
+    });
+    if (!result) return res.status(400).json({ message: 'Client not found' });
```
Comment (siva2204) [inline src/controllers/clientController.ts:213]: no need to check this again

File: src/controllers/clientController.ts [hunk 5]
```diff
@@ -121,3 +132,136 @@ export const getClientDetailsByUser = async (
...
+    return res.status(200).json({ message: 'Client updated successfully.' });
+  } catch (error) {
+    return res.status(500).json({ message: 'Internal server error' });
+  }
+};
+
+//to generate client secret
+export const generateSecret = async (
```
Comment (siva2204) [inline src/controllers/clientController.ts:221]: better to rename this to generateNewSecret ?

**Review by yellasd — APPROVED** [review]

**Pull request merged**